scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1624,56 +1671,95 @@
+          psuedo-matches [{:task-metadata-seq task-metadata-seq}]
+          task-txns (map (fn [job metadata] (job->task-txn job metadata compute-cluster)) jobs task-metadata-seq)]
+      (launch-tasks-for-cluster compute-cluster psuedo-matches task-txns pool-name conn nil false)
+      (log-structured/info "Launched jobs in Kubernetes compute cluster" {:pool pool-name
+                                                                          :compute-cluster cc-name
+                                                                          :number-launched-jobs jobs}))))
+
 (defn distribute-jobs-to-kubernetes
   "Distributes considerable jobs across the compute clusters and launches in Kubernetes."
-  [conn considerable-jobs pool-name compute-clusters mesos-run-as-user job->acceptable-compute-clusters-fn]
+  [conn considerable-jobs pool-name mesos-run-as-user compute-clusters
+   compute-cluster->scheduling-capacity job->acceptable-compute-clusters-fn]
   (tracing/with-span
     [s {:name "scheduler.distribute-jobs-to-kubernetes" :tags {:component tracing-component-tag}}]
     (prom/with-duration
       prom/scheduler-distribute-jobs-to-kubernetes-duration {:pool pool-name}
       (timers/time!
        (timers/timer (metric-title "distribute-jobs-to-kubernetes-duration" pool-name))
-       (let [; NOTE: this is required to properly filter compute clusters to
-             ; those configured for this pool. This is done via the synthetic-pods
-             ; config, accessed via the autoscaling? method. In the future, this list
-             ; of pools should be slightly restructured to be listed one level higher,
-             ; on the compute-cluster-template config.
-             compute-clusters-for-pool (filter #(cc/autoscaling? % pool-name) compute-clusters)
-             launch-distributed-job-fn
-             (fn [[compute-cluster jobs]]
-               (let [task-metadata-seq (jobs->kubernetes-task-metadata jobs pool-name mesos-run-as-user compute-cluster)
-                     psuedo-matches [{:task-metadata-seq task-metadata-seq}]
-                     task-txns (map (fn [job metadata] (job->task-txn job metadata compute-cluster)) jobs task-metadata-seq)]
-                 (launch-tasks-for-cluster compute-cluster psuedo-matches task-txns pool-name conn nil false)))]
+       (let [distributor-fn (partial scheduling-capacity-constrained-job-distributor 
+                                     compute-cluster->scheduling-capacity)
+             kubernetes-jobs-launcher (create-kubernetes-jobs-launcher conn pool-name mesos-run-as-user)]
          (log-structured/info "Launching jobs in Kubernetes" {:pool pool-name})
-         (distribute-and-launch-jobs considerable-jobs pool-name compute-clusters-for-pool
-                                     job->acceptable-compute-clusters-fn launch-distributed-job-fn)
-         (log-structured/info "Launched jobs in Kubernetes" {:pool pool-name :number-launched-jobs (count considerable-jobs)}))))))
```

Comment:
Retain this long line after its all done, that we've launched everything?

Reply:
done